js/component-courses.js

```diff
@@ -1,2 +1,52 @@
 // export as global since we're not using a module system
-window.Courses = { template: '<div>Classes</div>' }
+window.Courses = Vue.component("Courses", {
+    name: "Courses",
+    data: function() {
+        return {
+            list: null,
+            err: null,
+            warn: false
+        };
+    },
+
+    /* Before this view is rendered, fetch the data from the php file */
+    mounted: function () {
+        /* Fetch the returned json object value. */
+        fetch("get_info.php")
+        .then(function (response) {
+            return response.json()
+        })
+        .then(data => {
+            // If the data type is courses as expected, set the list attribute
+            // in this component
+            if (data.TYPE == "courses") {
+                console.log(data.DATA);
+                this.list = data.DATA;
+            }
+            // If the data type is error, display the error message
+            else if (data.TYPE == "error") {
+                console.log("data.DATA is not an appropriate array!");
```

Comment:
You don't know that...You also print an error in another way, so no need for this. There is `console.warn` and `console.error` if you really want to send an error to the console.